aiplatform/snippets/embeddings_preview.go

```diff
@@ -27,38 +26,34 @@ import (
 	"google.golang.org/protobuf/types/known/structpb"
 )
 
-func embedTextsPreview(
-	apiEndpoint, project, model string, texts []string,
-	task string, dimensionality *int) ([][]float32, error) {
+func embedTextsPreview(project, location, model string) ([][]float32, error) {
 	ctx := context.Background()
 
+	apiEndpoint := fmt.Sprintf("%s-aiplatform.googleapis.com:443", location)
+	dimensionality := 5
+	texts := []string{"banana muffins? ", "banana bread? banana muffins?"}
+
 	client, err := aiplatform.NewPredictionClient(ctx, option.WithEndpoint(apiEndpoint))
 	if err != nil {
 		return nil, err
 	}
 	defer client.Close()
 
-	match := regexp.MustCompile(`^(\w+-\w+)`).FindStringSubmatch(apiEndpoint)
-	location := "us-central1"
-	if match != nil {
-		location = match[1]
-	}
 	endpoint := fmt.Sprintf("projects/%s/locations/%s/publishers/google/models/%s", project, location, model)
 	instances := make([]*structpb.Value, len(texts))
 	for i, text := range texts {
 		instances[i] = structpb.NewStructValue(&structpb.Struct{
 			Fields: map[string]*structpb.Value{
 				"content":   structpb.NewStringValue(text),
-				"task_type": structpb.NewStringValue(task),
+				"task_type": structpb.NewStringValue("QUESTION_ANSWERING"),
```

Comment:
See previous around hard-coding this string literal into the object initialization.